core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -206,20 +206,75 @@ public void start() {
             AtomicInteger attempt = new AtomicInteger(0);
             Unreliables.retryUntilSuccess(startupAttempts, () -> {
                 logger().debug("Trying to start container: {} (attempt {}/{})", image.get(), attempt.incrementAndGet(), startupAttempts);
-                tryStart(profiler.startNested("Container startup attempt"));
+                tryRun(profiler.startNested("Container startup attempt"));
                 return true;
             });
-
         } catch (Exception e) {
             throw new ContainerLaunchException("Container startup failed", e);
         } finally {
             profiler.stop().log();
         }
     }
 
-    private void tryStart(Profiler profiler) {
+    /**
+     * Restarts a running container.
+     *
+     * @throws ContainerNotStartedException if {@link #start()} hasn't been called first
+     */
+    public void restart() {
+        Profiler profiler = new Profiler("Container restart");
+        profiler.setLogger(logger());
+
+        if (null == containerId) {
```

Comment:
I would remove this check, IMO It's better for UX to not fail here and just start a new container

@rnorth WDYT?

Reply:
I can do that, it's not a problem. @rnorth please let me know if you'd prefer this functionality.

Reply:
@rnorth ping